src/pint/models/glitch.py

```diff
@@ -145,43 +145,41 @@ def setup(self):
         ]
         for idx in set(self.glitch_indices):
             for param in self.glitch_prop:
-                if not hasattr(self, param + "%d" % idx):
+                if not hasattr(self, param + str(idx)):
                     param0 = getattr(self, f"{param}1")
                     self.add_param(param0.new_param(idx))
-                    getattr(self, param + "%d" % idx).value = 0.0
+                    getattr(self, param + str(idx)).value = 0.0
                 self.register_deriv_funcs(
-                    getattr(self, f"d_phase_d_{param[:-1]}"), param + "%d" % idx
+                    getattr(self, f"d_phase_d_{param[:-1]}"), param + str(idx)
                 )
```

Comment:
It will be better to use fstring instead of the + operator here since that is the style we had decided to prefer.